server/storage/memory.go

```diff
@@ -68,6 +75,17 @@ func (st *MemStorage) GetCurrent(gun, role string) (data []byte, err error) {
 	return space[len(space)-1].data, nil
 }
 
+// GetChecksum returns the metadada for a given role, under a GUN
+func (st *MemStorage) GetChecksum(gun, role, checksum string) (data []byte, err error) {
+	st.lock.Lock()
+	defer st.lock.Unlock()
+	data, ok := st.checksums[checksum]
+	if !ok || len(data) == 0 {
+		return nil, ErrNotFound{}
+	}
+	return data, nil
+}
+
 // Delete delets all the metadata for a given GUN
 func (st *MemStorage) Delete(gun string) error {
```

Comment:
Agreed, I think it's necessary for our memory store implementation since the tufMeta and checksum maps are separate.